### PR TITLE
fix/connectAuthorised wasn't returning the same instance as it should

### PR DESCRIPTION
**Pull request by bochaco**

### CODE REVIEW

File: src/api/app.js [hunk 1]
```diff
@@ -69,10 +69,7 @@ module.exports.authorise = (appToken, permissions, options) => {
 module.exports.connectAuthorised = (appToken, authUri) => {
   return getObj(appToken)
     .then((app) => app.auth.loginFromURI(authUri))
-    .then((connectedApp) => {
-      freeObj(appToken);
-      return genHandle(connectedApp);
-    });
+    .then((connectedApp) => appToken);
```
Comment (shankarsengalani) [inline src/api/app.js:72]: safe_app_nodejs doesn't return the same instance while calling `loginFromURI` API. [here](https://github.com/maidsafe/safe_app_nodejs/blob/master/src/app.js#L173) It creates a new app instance and returns that back.

Comment (bochaco) [reply src/api/app.js:72]: That's right, but both instances seem to reference the same connection, we can double check with Ben, but if this is correct, I thought it's better to return just the same reference/token to make it simpler for the user/client app.
For example, if you authorise an app (token/instance 1), and then connect as authorised (token/instance 2), you can check using either of the two tokens 1 or 2 if it's registered (using `window.safeApp.isRegistered(token)`) and both return `true`.

Comment (bochaco) [reply src/api/app.js:72]: ...or perhaps there is a bug in the safe-app and what I'm describing shouldn't occur.

Comment (gnunicorn) [reply src/api/app.js:72]: @shankar2105 you've referenced the wrong piece of code. The code is calling https://github.com/maidsafe/safe_app_nodejs/blob/master/src/index.js#L58 -- which isn't creating any new instance of SAFEApp but only mutates the existing one (by replacing the connection to the remote Rust object -- see https://github.com/maidsafe/safe_app_nodejs/blob/master/src/app.js#L135 ).

The function you have referenced is/can only be used when there isn't any SAFEApp instance yet -- thus it only ever creates but never _replaces_ any existing instance.

Comment (bochaco) [reply src/api/app.js:72]: Thanks a lot @gnunicorn .
That other function you referenced @shankar2105 , we could eventually also expose it as a separate function so the client app can connect using the URI and the appInfo without needing to call `initialise` and `connectAuthorised` but just call this function.
So I think we should be good to apply this fix now.

**Pull request merged**